wa/commands/record.py

```diff
@@ -49,46 +50,132 @@ class RecordCommand(Command):
+            self.manual_record(args)
+
+        self.revent_recorder.remove()
+
+    def record(self, revent_file, name, output_path):
+        msg = 'Press Enter when you are ready to record {}...'
+        self.logger.info(msg.format(name))
+        raw_input('')
+        self.revent_recorder.start_record(revent_file)
+        msg = 'Press Enter when you have finished recording {}...'
+        self.logger.info(msg.format(name))
+        raw_input('')
+        self.revent_recorder.stop_record()
+
+        output_path = output_path or os.getcwdu()
+        if not os.path.isdir(output_path):
+            os.makedirs(output_path)
+
+        revent_file_name = self.target.path.basename(revent_file)
+        host_path = os.path.join(output_path, revent_file_name)
+        if os.path.exists(host_path):
+            msg = 'Revent file \'{}\' already exists, overwrite? [y/n]'
+            self.logger.info(msg.format(revent_file_name))
+            if raw_input('') == 'y':
+                os.remove(host_path)
+            else:
+                msg = 'Did not pull and overwrite \'{}\''
+                self.logger.warning(msg.format(revent_file_name))
+                return
+        msg = 'Pulling \'{}\' from device'
+        self.logger.info(msg.format(self.target.path.basename(revent_file)))
+        self.target.pull(revent_file, output_path, as_root=self.target.is_rooted)
+
+    def manual_record(self, args):
+        file_name = '{}.{}revent'.format(self.target.model, args.suffix or '')
+        revent_file = self.target.get_workpath(file_name)
+        self.record(revent_file, '', args.output)
+
+    def package_record(self, args):
+        if args.clear:
+            self.target.execute('pm clear {}'.format(args.package))
+        self.logger.info('Starting {}'.format(args.package))
+        cmd = 'monkey -p {} -c android.intent.category.LAUNCHER 1'
+        self.target.execute(cmd.format(args.package))
+
+        file_name = '{}.{}revent'.format(self.target.model, args.suffix or '')
+        revent_file = self.target.get_workpath(file_name)
+        self.record(revent_file, '', args.output)
+
+    def workload_record(self, args):
+        context = LightContext(self.tm)
+        setup_revent = '{}.setup.revent'.format(self.target.model)
+        run_revent = '{}.run.revent'.format(self.target.model)
+        setup_file = self.target.get_workpath(setup_revent)
+        run_file = self.target.get_workpath(run_revent)
+
+        self.logger.info('Deploying {}'.format(args.workload))
+        workload = pluginloader.get_workload(args.workload, self.target)
+        workload.apk.init_resources(context.resolver)
+        workload.apk.setup(context.resolver)
+        sleep(workload.loading_time)
+
+        output_path = args.output or os.path.join(workload.dependencies_directory,
+                                                  'revent_files')
+        self.record(setup_file, 'SETUP', output_path)
+        self.record(run_file, 'RUN', output_path)
```

Comment:
`setup` recording should be optional. There should also be optional stages for `extract_results` and `teardown`.

The user should be able to specify with parameters which stages, in addition to `run`, they wish to record. E.g. -s for setup, -e for extract, -t for teardown, -a for all.